### PR TITLE
Extract Health- and Infocontroller

### LINKED ISSUE #66: Extract Health and Info-Controller
we should extract the logic for health and info-controller into a sperate package.
like https://github.com/ziyasal/hapijs-status-monitor

### PR DESCRIPTION
**Pull request by chrishelgert**
fixes #66. 

Extracted the health and Infocontroller in two middlewares (honeycomb-info-middleware, honeycomb-status-middleware)

### CODE REVIEW

File: packages/honeycomb-health-middleware/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+server.register([{
+  register: require('honeycomb-health-middleware'),
+}], (err) => {
+  if (err) {
+    throw err;
+  }
+
+  // ...
+});
+```
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/README.md]: Cleanup 😉 

```javascript
server.register([{
  register: require('honeycomb-health-middleware'),
}], (err) => {
  if (err) {
    throw err;
  }
});
```

File: packages/honeycomb-health-middleware/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "description": "honeycomb middleware for health-endpoint in hapi.js",
+  "main": "lib/index.js",
+  "scripts": {
+    "lint": "eslint --format=node_modules/eslint-formatter-pretty .",
+    "test": "npm run lint && npm run test:unit",
+    "test:unit": "jest --coverage"
+  },
+  "repository": {
+    "type": "git",
+    "url": "github.com/danielbayerlein/honeycomb"
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/package.json]:
```json
"url": "https://github.com/danielbayerlein/honeycomb.git"
```

File: packages/honeycomb-health-middleware/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "honeycomb-health-middleware",
+  "version": "1.0.0",
+  "description": "honeycomb middleware for health-endpoint in hapi.js",
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/package.json]:
```json
"description": "Health endpoint for Honeycomb",
```

File: packages/honeycomb-health-middleware/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# honeycomb-health-middleware
+
+> Health-Endpoint for Honeycomb
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/README.md]:
```
> Health endpoint for Honeycomb
```

File: packages/honeycomb-info-middleware/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# honeycomb-info-middleware
+
+> Info-Endpoint for Honeycomb
```
Comment (danielbayerlein) [inline packages/honeycomb-info-middleware/README.md]:
```
> Info endpoint for Honeycomb
```

File: packages/honeycomb-info-middleware/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    process,
+  },
+}], (err) => {
+  if (err) {
+    throw err;
+  }
+
+  // ...
+});
+```
```
Comment (danielbayerlein) [inline packages/honeycomb-info-middleware/README.md]: Cleanup 😉 

```javascript
server.register([{
  register: require('honeycomb-info-middleware'),
  options: {
    pkg: require('./package.json'),
    process,
  },
}], (err) => {
  if (err) {
    throw err;
  }
});
```

File: packages/honeycomb-health-middleware/__tests__/lib/index.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    register(server, options, next);
+
+    const call = server.route.mock.calls[0][0];
+
+    expect(call.method).toBe('GET');
+    expect(call.path).toBe('/health');
+    expect(typeof call.handler).toBe('function');
+  });
+
+  it('the health-route should return on object with status: up', () => {
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/__tests__/lib/index.js]: Typo `the health route should return an object with status`

File: packages/honeycomb-info-middleware/package.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "honeycomb-info-middleware",
+  "version": "1.0.0",
+  "description": "honeycomb middleware for info-endpoint in hapi.js",
```
Comment (danielbayerlein) [inline packages/honeycomb-info-middleware/package.json]:
```json
"description": "Info endpoint for Honeycomb",
```

File: packages/honeycomb-info-middleware/package.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  "main": "lib/index.js",
+  "scripts": {
+    "lint": "eslint --format=node_modules/eslint-formatter-pretty .",
+    "test": "npm run lint && npm run test:unit",
+    "test:unit": "jest --coverage"
+  },
+  "repository": {
+    "type": "git",
+    "url": "github.com/danielbayerlein/honeycomb"
```
Comment (danielbayerlein) [inline packages/honeycomb-info-middleware/package.json]:
```json
"url": "https://github.com/danielbayerlein/honeycomb.git"
```

**Comment by danielbayerlein** [comment]
@chrishelgert Please add the new packages to https://github.com/danielbayerlein/honeycomb/blob/master/.travis.yml#L4

File: packages/generator-honeycomb/app/templates/.config/_server.js [hunk 10]
```diff
@@ -70,6 +70,20 @@ const defaultConfig = {
         },
       },
     },
+    {
+      plugin: {
+        register: 'honeycomb-health-middleware',
```
Comment (chrishelgert) [inline packages/generator-honeycomb/app/templates/.config/_server.js]: yes it works 👍

File: packages/honeycomb-health-middleware/README.md (hunk 4 above)
Comment (chrishelgert) [inline packages/honeycomb-health-middleware/README.md]: check

File: packages/honeycomb-health-middleware/README.md (hunk 1 above)
Comment (chrishelgert) [inline packages/honeycomb-health-middleware/README.md]: check

File: packages/honeycomb-health-middleware/package.json (hunk 3 above)
Comment (chrishelgert) [inline packages/honeycomb-health-middleware/package.json]: check

File: packages/honeycomb-health-middleware/package.json (hunk 2 above)
Comment (chrishelgert) [inline packages/honeycomb-health-middleware/package.json]: check

File: packages/honeycomb-info-middleware/README.md (hunk 5 above)
Comment (chrishelgert) [inline packages/honeycomb-info-middleware/README.md]: check

File: packages/honeycomb-info-middleware/README.md (hunk 6 above)
Comment (chrishelgert) [inline packages/honeycomb-info-middleware/README.md]: check

File: packages/honeycomb-health-middleware/__tests__/lib/index.js (hunk 7 above)
Comment (chrishelgert) [inline packages/honeycomb-health-middleware/__tests__/lib/index.js]: check

File: packages/honeycomb-info-middleware/package.json (hunk 8 above)
Comment (chrishelgert) [inline packages/honeycomb-info-middleware/package.json]: check

File: packages/honeycomb-info-middleware/package.json (hunk 9 above)
Comment (chrishelgert) [inline packages/honeycomb-info-middleware/package.json]: check

**Comment by chrishelgert** [comment]
@danielbayerlein changes are done 😸

File: packages/honeycomb-info-middleware/package.json [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  "version": "1.0.0",
+  "description": "Info endpoint for Honeycomb",
+  "main": "lib/index.js",
+  "scripts": {
+    "lint": "eslint --format=node_modules/eslint-formatter-pretty .",
+    "test": "npm run lint && npm run test:unit",
+    "test:unit": "jest --coverage"
+  },
+  "repository": {
+    "url": "https://github.com/danielbayerlein/honeycomb.git"
```
Comment (danielbayerlein) [inline packages/honeycomb-info-middleware/package.json]: `type` not necessary? See https://docs.npmjs.com/files/package.json#repository

File: packages/honeycomb-health-middleware/package.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  "version": "1.0.0",
+  "description": "Health endpoint for Honeycomb",
+  "main": "lib/index.js",
+  "scripts": {
+    "lint": "eslint --format=node_modules/eslint-formatter-pretty .",
+    "test": "npm run lint && npm run test:unit",
+    "test:unit": "jest --coverage"
+  },
+  "repository": {
+    "url": "https://github.com/danielbayerlein/honeycomb.git"
```
Comment (danielbayerlein) [inline packages/honeycomb-health-middleware/package.json]: `type` not necessary? See https://docs.npmjs.com/files/package.json#repository

Comment (chrishelgert) [inline packages/honeycomb-health-middleware/package.json]: ups dropped one line to much 😞 ,
pushed it with type again

**Comment by danielbayerlein** [comment]
Thanks ❤️

**Pull request merged**